runtime/lua/vim/cache.lua

```diff
@@ -0,0 +1,481 @@
+-- interop with the native Neovim cache
+if type(package.loaded['vim.cache']) == 'table' then
+  return package.loaded['vim.cache']
+end
+
+local ffi = require('ffi')
```

Comment:
This means this only works with luajit, right?

Reply:
Yes, we need to remove this.

Reply:
Or disable the cache for PUC Lua builds? If you build against that, you obviously do not care about speed at all...

Reply:
It's not needed though.

Reply:
Since Lua 5.1 is a strictly defined language and LuaJIT is just an implementation, I'm quite keen on the idea of us being able to move across to any 5.1 implementation that comes about.

Reply:
@lewis6991 i wouldn't use mpack for the file-based cache.

I think it's better to just not enable the cache when `ffi` is not available. Will change it.

Reply:
> i wouldn't use mpack for the file-based cache.

Is there a technical reason why?

Reply:
Alright, nevermind. Will change the implementation so it no longer needs ffi and also won't need mpack

Reply:
@lewis6991 performance would be greatly impacted. But I can prevent using either. Working on a better solution.

Reply:
How? It's simply used to read/write a single header.

How greatly are we talking?

Reply:
> except to show PUC Lua builders the error of their ways :]

Yes, how dare people run nvim on platforms that LuaJIT doesn't support!

Reply:
Exactly!

Reply:
Just changed the implementation so it no longer needs ffi (or mpack). Performance is the same.